server/websocket.go

```diff
@@ -650,6 +651,16 @@ func (p *Plugin) handleJoin(userID, connID, channelID, title, threadID string) (
 		}, &model.WebsocketBroadcast{ChannelId: channelID, ReliableClusterSend: true})
 	}
 
+	clientStateData, err := json.Marshal(state.Call.getClientState(p.getBotID(), userID))
+	if err != nil {
+		p.LogError("failed to marshal client state", "err", err.Error())
+	} else {
+		p.publishWebSocketEvent(wsEventCallState, map[string]interface{}{
+			"channel_id": channelID,
+			"call":       string(clientStateData),
```

Comment:
Unfortunately we need to do this horrible JSON in JSON trick since the generic map would otherwise fail to marshal out structure.

Reply:
Yes, but it would need to be MM server side change (gob encoding) which should then be aware of call specific types. Not sure I want to go there just yet.

Reply:
Ah yah, that's too much. Forget I asked. Maybe... Could we add `channel_id` to the call state, and then send the whole thing (no embedding)?

Reply:
We can only send a map, not custom data so the problem would be the same. It was a simple object we could probably create a custom marshaler to covert from struct to `map[string]any` but even then it's ugly.

Reply:
But if it's a map[string]interface{}, that can hold a single level struct that's been converted to json, and then you can use that directly as the object in the front end, right?  (I don't mean to delay the PR over it, I'm just trying to think how we can make this better, because sending over a string just feels wrong)

Reply:
@cpoile If you save it as a JSON string like we are doing yes. But otherwise we'll eventually try to marshal that in GOB and fail to do so because the plugin RPC layer doesn't have a marshaler for our type.

Reply:
I guess I was thinking of something like marshalling into a json blob, then unmarshalling into a map[string]interface{}, then sending that? Something along those lines. (Not sending a go struct.)

Reply:
Right, that would work for a simple struct but will break in our case because we'd have to do that "deeply", and check any substructs like userState, recordingState and such. It's doable but you need to have this monster marshaler, maybe even use reflect to make it actually helpful, not sure it's worth it.

Reply:
Yah, anyway, just thinking out loud. :)